util/app/discovery/discovery.go

```diff
@@ -159,3 +139,36 @@ func matchRepositoryCMP(ctx context.Context, repoPath string, client pluginclien
 	}
 	return resp.GetIsSupported(), nil
 }
+
+func cmpSupports(ctx context.Context, pluginSockFilePath, repoPath, fileName string, env []string, tarExcludedGlobs []string) (io.Closer, pluginclient.ConfigManagementPluginServiceClient, bool) {
+	address := filepath.Join(pluginSockFilePath, fileName)
```

Comment:
I think we need traversal protection on `fileName`. Looks like the filename must always end with `.sock`. Even so, an attacker might try different paths to see if they can get different errors for existing directories vs. non-existing directories.

Reply:
Not sure if we expose any directory traversal attacks, if the user explicitly specifies the plugin name we only check for the existence of `<pluginpath>/<plugin-name>.sock` so in all cases `.sock` gets appended and the user input is not used as is and also the only directory that's available is `pluginpath` which is only used for storing the plugin sock files as far as I can see. Maybe I am missing something but I do not see any way a user can check for exisiting/non-existing directories as we always use a file and not iterate directories when a name is given

Reply:
If the user provides `plugin name = "../test-dir/nothing", then the final path is `/test-dir/nothing.sock`. If the error message for a missing directory is different from a missing file, the attacker could infer whether `/test-dir` exists.

I haven't tested it. But either way, I think we should probably use something like the `files.Inbound` utility function just to be safe.

Reply:
I see have updated the code with `Inbound` check